### PR TITLE
AXAPI in python

### PR DESCRIPTION
**Pull request by alice**

### CODE REVIEW

File: tools/wptrunner/wptrunner/executors/protocol.py [hunk 1]
```diff
@@ -321,6 +321,20 @@ def get_computed_role(self, element):
...
 
+class PlatformAccessibilityProtocolPart(ProtocolPart):
+    """Protocol part for platform accessibility introspection"""
+    __metaclass__ = ABCMeta
+
+    name = "platform_accessibility"
+
+    @abstractmethod
+    def get_accessibility_api_node(self, dom_id):
+        """Return the the platform accessibilty object.
```
Comment (Ms2ger) [inline tools/wptrunner/wptrunner/executors/protocol.py:332]:
```suggestion
        """Return the the platform accessibility object.
```

File: tools/wptrunner/wptrunner/testrunner.py [hunk 2]
```diff
@@ -493,7 +494,7 @@ def wait_event(self):
         }
         try:
             command, data = self.command_queue.get(True, 1)
-            self.logger.debug("Got command: %r" % command)
+            #self.logger.debug("Got command: %r" % command)
```
Comment (Ms2ger) [inline tools/wptrunner/wptrunner/testrunner.py:497]:
```suggestion
            self.logger.debug("Got command: %r" % command)
```

File: tools/wptrunner/wptrunner/wptcommandline.py [hunk 3]
```diff
@@ -377,6 +377,8 @@ def create_parser(product_choices=None):
     chrome_group.add_argument("--no-enable-experimental", action="store_false", dest="enable_experimental",
                               help="Do not enable --enable-experimental-web-platform-features flag "
                               "on experimental channels")
+    chrome_group.add_argument( "--force-renderer-accessibility", action="store_true",
+                              dest="force_renderer_accessibility",help="Turn on accessibility.")
```
Comment (Ms2ger) [inline tools/wptrunner/wptrunner/wptcommandline.py:381]:
```suggestion
    chrome_group.add_argument("--force-renderer-accessibility", action="store_true",
                              dest="force_renderer_accessibility", help="Turn on accessibility.")
```

File: resources/testdriver.js [hunk 4]
```diff
@@ -1254,6 +1266,10 @@
 
         async clear_device_posture(context=null) {
             throw new Error("clear_device_posture() is not implemented by testdriver-vendor.js");
+        },
+
+        async get_accessibility_api_node(dom_id) {
+            throw new Error("not implemented, whoops!");
```
Comment (Ms2ger) [inline resources/testdriver.js:1272]:
```suggestion
            throw new Error("get_accessibility_api_node() is not implemented by testdriver-vendor.js");
```

File: tools/wptrunner/wptrunner/executors/executoracacia.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+from sys import platform
+linux = False
+mac = False
+if platform == "linux":
+  linux = True
+  from .executoratspi import *
+if platform == "darwin":
+  mac = True
+  from .executoraxapi import *
```
Comment (Ms2ger) [inline tools/wptrunner/wptrunner/executors/executoracacia.py:11]: Four spaces, and did you forget to add these files?

File: tools/wptrunner/wptrunner/executors/executoracacia.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+class AcaciaPlatformAccessibilityProtocolPart(PlatformAccessibilityProtocolPart):
+    def setup(self):
+        self.product_name = self.parent.product_name
+        self.impl = None
+        if linux:
+            self.impl = AtspiExecutorImpl()
+            self.impl.setup(self.product_name)
+        if mac:
+            self.impl = AXAPIExecutorImpl()
+            self.impl.setup(self.product_name)
```
Comment (Ms2ger) [inline tools/wptrunner/wptrunner/executors/executoracacia.py:23]: What happens on Windows?

**Comment by spectranaut** [comment]
Whooops, thanks for the review @Ms2ger :) -- but this was suppose to be a PR on the Igalia branch against Igalia main for me to take a look at, it's still a proof of concept/exploration not ready for WPT review

**Pull request closed without merging**